sonarr/series.go

```diff
@@ -27,50 +105,52 @@ func (s *Sonarr) GetSeries(tvdbID int64) ([]*Series, error) {
 }
 
 func (s *Sonarr) GetSeriesContext(ctx context.Context, tvdbID int64) ([]*Series, error) {
-	params := make(url.Values)
+	var output []*Series
 
+	params := make(url.Values)
 	if tvdbID != 0 {
 		params.Add("tvdbId", strconv.FormatInt(tvdbID, starr.Base10))
 	}
 
-	var series []*Series
-
-	_, err := s.GetInto(ctx, "v3/series", params, &series)
-	if err != nil {
+	if _, err := s.GetInto(ctx, bpSeries, params, &output); err != nil {
 		return nil, fmt.Errorf("api.Get(series): %w", err)
 	}
 
-	return series, nil
+	return output, nil
 }
 
 // UpdateSeries updates a series in place.
-func (s *Sonarr) UpdateSeries(seriesID int64, series *Series) error {
-	return s.UpdateSeriesContext(context.Background(), seriesID, series)
+func (s *Sonarr) UpdateSeries(series *Series) (*Series, error) {
+	return s.UpdateSeriesContext(context.Background(), series)
 }
 
-func (s *Sonarr) UpdateSeriesContext(ctx context.Context, seriesID int64, series *Series) error {
+func (s *Sonarr) UpdateSeriesContext(ctx context.Context, series *Series) (*Series, error) {
+	var output *Series
+
 	params := make(url.Values)
 	params.Add("moveFiles", "true")
 
 	var body bytes.Buffer
 	if err := json.NewEncoder(&body).Encode(series); err != nil {
-		return fmt.Errorf("json.Marshal(series): %w", err)
+		return nil, fmt.Errorf("json.Marshal(series): %w", err)
 	}
 
-	_, err := s.Put(ctx, "v3/series/"+strconv.FormatInt(seriesID, starr.Base10), params, &body)
-	if err != nil {
-		return fmt.Errorf("api.Put(series): %w", err)
+	uri := path.Join(bpSeries, strconv.Itoa(int(series.ID)))
+	if _, err := s.PutInto(ctx, uri, params, &body, &output); err != nil {
+		return nil, fmt.Errorf("api.Put(series): %w", err)
```

Comment:
A new pattern I'm going to do is this:
```suggestion
		return nil, fmt.Errorf("api.Put(%s): %w", uri, err)
```

Reply:
ok, I'll change that